e2e/vmss.go

```diff
@@ -232,6 +249,94 @@ func getVmssName(t *testing.T) string {
 	return name
 }
 
+func getBaseVMSSModelSelfContained(name, sshPublicKey string, opts *scenarioRunOpts) armcompute.VirtualMachineScaleSet {
+	nbc, err := json.Marshal(baseNodeBootstrappingContract(config.Config.Location, opts))
+	if err != nil {
+		return armcompute.VirtualMachineScaleSet{}
+	}
+
+	nbcEntry := `#cloud-config
+
+write_files:
+- path: /opt/azure/containers/nbc.json
+  permissions: "0755"
+  owner: root
+  content: !!binary |
+    %s`
+	customData := fmt.Sprintf(nbcEntry, base64.StdEncoding.EncodeToString(nbc))
+	encodedCustomData := base64.StdEncoding.EncodeToString([]byte(customData))
+	return armcompute.VirtualMachineScaleSet{
+		Location: to.Ptr(config.Config.Location),
+		SKU: &armcompute.SKU{
+			Name:     to.Ptr("Standard_D2ds_v5"),
+			Capacity: to.Ptr[int64](1),
+		},
+		Properties: &armcompute.VirtualMachineScaleSetProperties{
+			Overprovision: to.Ptr(false),
+			UpgradePolicy: &armcompute.UpgradePolicy{
+				Mode: to.Ptr(armcompute.UpgradeModeManual),
+			},
+			VirtualMachineProfile: &armcompute.VirtualMachineScaleSetVMProfile{
+				OSProfile: &armcompute.VirtualMachineScaleSetOSProfile{
+					ComputerNamePrefix: to.Ptr(name),
+					AdminUsername:      to.Ptr("azureuser"),
+					CustomData:         &encodedCustomData,
```

Comment:
so I guess only customdata?

Reply:
yep, will add the cse script to poll bootstrap status in separate pr